presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeJoinQueries.java

```diff
@@ -13,21 +13,95 @@
  */
 package com.facebook.presto.nativeworker;
 
+import com.facebook.presto.execution.SqlQueryManager;
+import com.facebook.presto.spi.Plugin;
+import com.facebook.presto.spi.statistics.HistoryBasedPlanStatisticsProvider;
+import com.facebook.presto.sql.planner.plan.JoinNode;
 import com.facebook.presto.testing.ExpectedQueryRunner;
+import com.facebook.presto.testing.InMemoryHistoryBasedPlanStatisticsProvider;
 import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.any;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.node;
 
 public class TestPrestoNativeJoinQueries
         extends AbstractTestNativeJoinQueries
 {
     @Override
     protected QueryRunner createQueryRunner() throws Exception
     {
-        return PrestoNativeQueryRunnerUtils.createNativeQueryRunner(true);
+        QueryRunner queryRunner = PrestoNativeQueryRunnerUtils.createNativeQueryRunner(true);
+        queryRunner.installPlugin(new Plugin()
+        {
+            @Override
+            public Iterable<HistoryBasedPlanStatisticsProvider> getHistoryBasedPlanStatisticsProviders()
+            {
+                return ImmutableList.of(new InMemoryHistoryBasedPlanStatisticsProvider());
+            }
+        });
+        return queryRunner;
     }
 
     @Override
     protected ExpectedQueryRunner createExpectedQueryRunner() throws Exception
     {
         return PrestoNativeQueryRunnerUtils.createJavaQueryRunner();
     }
+
+    @Test
+    public void testJoinNull()
+    {
+        String sql = "SELECT\n" +
```

Comment:
@pranjalssh Thank you for adding a test.

This is a large query. Wondering, what's the motivation for using something so complex vs. a simple join? 

```
SELECT count(1) 
FROM orders o LEFT JOIN lineitem l 
   ON if(o.orderkey % 2 = 0, o.orderkey) = if(l.orderkey % 11 = 0, l.orderkey)
```